spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/interpreter/rules/FillNullsRuleInterpreter.scala

```diff
@@ -41,22 +44,33 @@ case class FillNullsRuleInterpreter(rule: FillNullsConformanceRule) extends Rule
       rule.outputColumn
     )
 
+    val dataType = SchemaUtils.getFieldType(rule.inputColumn, df.schema).get
+    val default: Column = simpleLiteralCast(rule.value, dataType) match {
+      case Success(value) => value
+      case Failure(_) =>
+        log.warn(
+          s"""Unable to cast literal ${rule.value} to $dataType
+             |for FillNulls conformance rule number ${rule.order}.
+             |Using string as a fallback.""".stripMargin.replaceAll("[\\r\\n]", ""))
+        lit(rule.value)
+    }
+
     if (rule.outputColumn.contains('.')) {
-      conformNestedField(df)
+      conformNestedField(df, default)
     } else {
-      conformRootField(df)
+      conformRootField(df, default)
     }
   }
 
-  private def conformNestedField(df: Dataset[Row])(implicit spark: SparkSession): Dataset[Row] = {
+  private def conformNestedField(df: Dataset[Row], default: Column)(implicit spark: SparkSession): Dataset[Row] = {
     df.nestedWithColumnExtended(rule.outputColumn, getField => when(
-      getField(rule.inputColumn).isNull, inferStrictestType(rule.value)
+      getField(rule.inputColumn).isNull, default
```

Comment:
Alternatively, there is `coalesce()` in Spark that does the same thing, just more concise code.

Reply:
Sorry missed this earlier. Implemented